terraform/modules/spack/opensearch.tf

```diff
@@ -1,28 +1,40 @@
 locals {
   domain_endpoint_name = "opensearch.${var.deployment_name == "prod" ? "" : "${var.deployment_name}."}spack.io"
-  cognito_enabled = var.deployment_name == "prod"
+  cognito_enabled      = var.deployment_name == "prod"
 }
 
-resource "aws_opensearch_domain" "spack" {
-  count = var.provision_opensearch_cluster ? 1 : 0
+resource "random_password" "opensearch_password" {
+  length  = 64
+  special = true
+}
 
+resource "aws_opensearch_domain" "spack" {
   domain_name = "spack${var.deployment_name == "prod" ? "" : "-${var.deployment_name}"}"
 
   engine_version = "OpenSearch_1.3"
 
   advanced_security_options {
     enabled                        = true
     internal_user_database_enabled = true
+
+    dynamic "master_user_options" {
+      for_each = var.deployment_name == "prod" ? [] : [1]
+
+      content {
+        master_user_name     = "admin"
+        master_user_password = random_password.opensearch_password.result
+      }
```

Comment:
We use the existing production password in some places, such as https://github.com/spack/spack-infrastructure/blob/main/images/gitlab-error-processor/job-template.yaml#L54. So, we'll need to be careful of that. Since we're specifying `length = 64` and `special = true`, I'm not sure if the existing prod password is compatible.